viadot/sources/base.py

```diff
@@ -156,10 +156,12 @@ def to_parquet(
             out_df = df
 
         # create directories if they don't exist
-
-        if not os.path.isfile(path):
-            directory = os.path.dirname(path)
-            os.makedirs(directory, exist_ok=True)
+        try:
+            if not os.path.isfile(path):
+                directory = os.path.dirname(path)
+                os.makedirs(directory, exist_ok=True)
+        except FileNotFoundError:
+            pass
```

Comment:
Just a small suggestion. What do you think about adding a logger there with the information that the file is not found?  To catch the exception and provide some information.